GrocifyApp.UI/src/App.jsx

```diff
@@ -25,6 +27,36 @@ PrivateRoute.propTypes = {
 
 function App() {
     const [isDarkMode, setDarkMode] = useState(false);
+    const [userAuth, setUserAuth] = useState(null);
+    const userId = "581ccc32-dd5d-455b-d2c2-08dc11ed02ad";
+    const [errorMessage, setErrorMessage] = useState('');
+
+    useEffect(() => {
+        getUserDarkMode();
+    }, []);
+
+    const getUserDarkMode = async () => {
+        try {
+            const response = await fetch(`api/User/${userId}`, {
```

Comment:
remove this call to API. when you do the login, you have the information of the user, with DarkMode. So, you can save darkmode in localStorage, as you do for token. And have no need to always make this call at the beggining, what do you think?